File: Tarea 4 - La redencion/racing.py
#Juan Miguel Rojas Mejía
#8963761

from sys import stdin
from collections import deque

def solve(tianJiL, kingL, N):
    tianJi = deque()
    king = deque ()
    tianJi.extend(tianJiL)
    king.extend(kingL)
    ans = 0
    while(N != 0):
        if(tianJi[len(tianJi)-1] > king[len(king)-1]):
            ans+=200
            tianJi.pop()
            king.pop()
        elif(tianJi[len(tianJi)-1] < king[len(king)-1]):
            ans-=200
            king.pop()
            tianJi.popleft()
        else:
            if(tianJi[0] > king[0]):
                ans+=200
                tianJi.popleft()
                king.popleft()
            elif(king[len(tianJi)-1] != tianJi[0]):
                ans-=200
                king.pop()
                tianJi.popleft()
        N-=1

    return ans

def main():
    N = int(stdin.readline())
    val = 0
    while(N != 0):
        tianJi = list(map(int, stdin.readline().split()))
        king = list(map(int, stdin.readline().split()))
        tianJi.sort()
        king.sort()
        print(solve(tianJi, king, N))
        N = int(stdin.readline())

main()

File: Tarea 4 - La redencion/racing.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
#include <deque>
using namespace std;

void convert(vector<int>& tianJi, vector<int>& king, deque<int>& tianQueue, deque<int>& kingQueue, int N){
    for(int i = 0; i < N; i++) tianQueue.push_back(tianJi[i]);
    for(int i = 0; i < N; i++) kingQueue.push_back(king[i]);
}

int solve(vector<int>& tianJi, vector<int>& king, int N){
    int ans = 0;
    deque<int> tianQueue;
    deque<int> kingQueue;
    convert(tianJi, king, tianQueue, kingQueue, N);
    while(N--){
        if(tianQueue.back() > kingQueue.back()){
            ans+=200;
            tianQueue.pop_back();
            kingQueue.pop_back();
        }else if(tianQueue.back() < kingQueue.back()){
            ans-=200;
            kingQueue.pop_back();
            tianQueue.pop_front();
        }else{
            if(tianQueue.front() > kingQueue.front()){
                ans+=200;
                tianQueue.pop_front();
                kingQueue.pop_front();
            }else if(kingQueue.back() != tianQueue.front()){
                ans-=200;
                kingQueue.pop_back();
                tianQueue.pop_front();
            }
        }
    }
   
   return ans;
}

int main(){
    //freopen("racingin.txt", "r", stdin);
    //freopen("racingout.txt", "w", stdout);
    int N;
    cin >> N;
    while(N != 0){
        vector<int> tianJi(N);
        vector<int> king(N);
        for(int i = 0; i < N; i++) cin >> tianJi[i];
        for(int i = 0; i < N; i++) cin >> king[i];
        sort(tianJi.begin(), tianJi.end());
        sort(king.begin(), king.end());
        cout << solve(tianJi, king, N) << endl;
        cin >> N;
    }
    return 0;
}